src/app/devices/deviceContent/components/shared/modelDefinitionSource.tsx

```diff
@@ -0,0 +1,52 @@
+/***********************************************************
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License
+ **********************************************************/
+import * as React from 'react';
+import { Stack } from 'office-ui-fabric-react/lib/Stack';
+import { ActionButton } from 'office-ui-fabric-react/lib/Button';
+import { Label } from 'office-ui-fabric-react/lib/Label';
+import { useLocalizationContext } from '../../../../shared/contexts/localizationContext';
+import { REPOSITORY_LOCATION_TYPE } from '../../../../constants/repositoryLocationTypes';
+import { ResourceKeys } from '../../../../../localization/resourceKeys';
+import '../../../../css/_modelDefinitionSource.scss';
+
+interface ModelDefinitionSourceViewProps {
+    source: REPOSITORY_LOCATION_TYPE;
+    handleConfigure: () => void;
+}
+
+const getModelDefinitionSourceResourceKeys = (source: string) => {
+    switch (source) {
+        case REPOSITORY_LOCATION_TYPE.Public:
+            return ResourceKeys.settings.modelDefinitions.repositoryTypes.public.label;
```

Comment:
nit: could have been written as `ResourceKeys.settings.modelDefinitions.repositoryTypes[`#{source}`].label instead of switch case